src/nnet3/nnet-compute-test.cc

```diff
@@ -27,6 +28,51 @@
 namespace kaldi {
 namespace nnet3 {
 
+void UnitTestNnetComputationIo(NnetComputation *computation) {
+  bool binary = (Rand() % 2 == 0);
+  std::ostringstream os;
+  computation->Write(os, binary);
+  const std::string &original_output = os.str();
+  std::istringstream computation_is(original_output);
+  computation->Read(computation_is, binary);
+  std::istringstream computation_is2(original_output);
+  NnetComputation computation2;
+  computation2.Read(computation_is2, binary);
+    
+  std::ostringstream os2, os3;
+  computation->Write(os2, binary);
+  computation2.Write(os3, binary);
+  
+  if (binary) {
+    KALDI_LOG << os2.str();
+    KALDI_LOG << original_output;
+    KALDI_ASSERT(os2.str() == original_output);
+    KALDI_ASSERT(os3.str() == original_output);
+  }
+}
+
+void UnitTestComputationRequestIo(ComputationRequest *request) {
+  bool binary = (Rand() % 2 == 0);
+  std::ostringstream os;
+  request->Write(os, binary);
+  const std::string &original_output = os.str();
+  std::istringstream request_is(original_output);
+  request->Read(request_is, binary);
+  std::istringstream request_is2(original_output);
+  ComputationRequest request2;
+  request2.Read(request_is2, binary);
+    
+  std::ostringstream os2, os3;
+  request->Write(os2, binary);
+  request2.Write(os3, binary);
+  
+  if (binary) {
+    KALDI_LOG << os2.str();
+    KALDI_LOG << original_output;
+    KALDI_ASSERT(os2.str() == original_output);
+    KALDI_ASSERT(os3.str() == original_output);
```

Comment:
since the object has, I think, an operator ==, why not use that as an additional test here?  Do the same in any similar situations.